example/Index.re

```diff
@@ -1,11 +1,24 @@
 open Express;
 
-let __dirname: option string = [%bs.node __dirname];
+let app = express ();
 
-let app = Express.express ();
+App.useOnPath app path::"/" @@ Middleware.fromF (fun _ _ next => {
+  Js.log "Request received"; 
+     /* This will be printed for every request */
+  next Js.undefined [@bs]
```

Comment:
If we drop the `@bs` then yes you will get a Curry call. Similarly if you have a function for the undefined (which I had at first) then you will have a reference to that module rather than more idiomatic JS. 

Both of them are there to improve generated code (and a bit performance but I don't think it's that crucial here). I don't have a strong opinion.